src/map.rs

```diff
@@ -1307,25 +1294,25 @@ where
         // there are three cases when a bin pointer is invalidated:
         //
         //  1. if the table was resized, bin is a move entry, and the resize has completed. in
-        //     that case, the table (and all its heads) will be dropped in the next epoch
-        //     following that.
+        //     that case, the table (and all its heads) will be retired.
         //  2. if the table is being resized, bin may be swapped with a move entry. the old bin
-        //     will then be dropped in the following epoch after that happens.
+        //     will be retired.
         //  3. when elements are inserted into or removed from the map, bin may be changed into
         //     or from a TreeBin from or into a regular, linear bin. the old bin will also be
-        //     dropped in the following epoch if that happens.
+        //     retired.
         //
         // in all cases, we held the guard when we got the reference to the bin. if any such
         // swap happened, it must have happened _after_ we read. since we did the read while
-        // pinning the epoch, the drop must happen in the _next_ epoch (i.e., the one that we
-        // are holding up by holding on to our guard).
+        // the current thread was marked as active, we must be included in the reference count,
+        // and the drop must happen _after_ we decrement the count (i.e drop our guard).
         let node = table.find(unsafe { bin.deref() }, h, key, guard);
         if node.is_null() {
             return None;
         }
-        // safety: we read the bin while pinning the epoch. a bin will never be dropped until the
-        // next epoch after it is removed. since it wasn't removed, and the epoch was pinned, that
-        // cannot be until after we drop our guard.
+
+        // safety: we loaded the bin while holding a guard, so any retirements
```

Comment:
_and_ any _future_ retirements must see us as active

Reply:
Fixed in f24162f1abd3f954060eff5026e961a95a876231